plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSnapshots.java

```diff
@@ -0,0 +1,84 @@
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static java.lang.String.format;
+
+public class TestIcebergSnapshots
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return IcebergQueryRunner.createIcebergQueryRunner(ImmutableMap.of(), new IcebergConfig().getFileFormat(), ImmutableList.of());
+    }
+
+    @Test
+    public void testReadingFromSpecificSnapshot()
+    {
+        String tableName = "test_reading_snapshot" + randomTableSuffix();
+        assertUpdate(format("CREATE TABLE %s (a bigint, b bigint)", tableName));
+        assertUpdate(format("INSERT INTO %s VALUES(1, 1)", tableName), 1);
+        List<MaterializedRow> materializedRows = getQueryRunner().execute(format("SELECT snapshot_id, parent_id FROM \"%s$snapshots\"", tableName)).getMaterializedRows();
+        MaterializedRow oldest = getOldest(materializedRows);
+        long oldestId = (long) oldest.getField(0);
+        long childId = (long) getChild(materializedRows, oldestId).getField(0);
+
+        assertQuery(format("SELECT count(*) FROM \"%s@%d\"", tableName, oldestId), "VALUES(0)");
+        assertQuery(format("SELECT * FROM \"%s@%d\"", tableName, childId), "VALUES(1,1)");
+        assertUpdate(format("DROP TABLE %s", tableName));
+    }
+
+    @Test
+    public void testSelectWithMoreThanOneSnapshotOfTheSameTable()
+    {
+        String tableName = "test_reading_snapshot" + randomTableSuffix();
+        assertUpdate(format("CREATE TABLE %s (a bigint, b bigint)", tableName));
+        assertUpdate(format("INSERT INTO %s VALUES(1, 1)", tableName), 1);
+        assertUpdate(format("INSERT INTO %s VALUES(2, 2)", tableName), 1);
+        List<MaterializedRow> materializedRows = getQueryRunner().execute(format("SELECT snapshot_id, parent_id FROM \"%s$snapshots\"", tableName)).getMaterializedRows();
+        MaterializedRow oldest = getOldest(materializedRows);
+        long oldestId = (long) oldest.getField(0);
+        long firstChildId = (long) getChild(materializedRows, oldestId).getField(0);
+
+        assertQuery(format("SELECT * FROM %s", tableName), "SELECT * FROM (VALUES(1,1), (2,2))");
+        assertQuery(format("SELECT * FROM (SELECT * FROM %s EXCEPT SELECT * FROM \"%s@%d\")", tableName, tableName, firstChildId), "VALUES(2,2)");
+        assertUpdate(format("DROP TABLE %s", tableName));
+    }
+
+    private MaterializedRow getOldest(List<MaterializedRow> rows)
+    {
+        return rows.stream()
+                .filter(x -> x.getField(1) == null)
+                .findFirst().get();
+    }
+
+    private MaterializedRow getChild(List<MaterializedRow> rows, long parentId)
+    {
+        return rows.stream()
+                .filter(x -> x.getField(1) != null && x.getField(1).equals(parentId))
+                .findFirst().get();
+    }
```

Comment:
those are cryptic.
Avoid single letter variables `x`.
What is `1`.
provide better method names.